modules/mockk/src/jvmMain/kotlin/io/mockk/impl/recording/JvmSignatureValueGenerator.kt

```diff
@@ -62,4 +62,27 @@ class JvmSignatureValueGenerator(val rnd: Random) : SignatureValueGenerator {
                 } as T
             }
     }
+
+    /**
+     * Generates a Long value in Duration's safe millisecond range to avoid AssertionError.
+     * Uses wider millisecond range while avoiding the denormalized range.
+     * See: https://github.com/mockk/mockk/issues/1401
+     */
+    private fun generateSafeLong(): Long {
+        return if (rnd.nextBoolean()) {
+            // Generate positive value in safe millisecond range (avoids denormalized)
+            rnd.nextLong(MAX_NANOS_IN_MILLIS + 1, MAX_MILLIS)
+        } else {
+            // Generate negative value in safe millisecond range (avoids denormalized)
+            rnd.nextLong(-MAX_MILLIS, -MAX_NANOS_IN_MILLIS)
+        }
+    }
+
+    companion object {
+        // Duration internal constants (mirrored from kotlin.time.Duration)
+        private const val NANOS_IN_MILLIS = 1_000_000L
```

Comment:
Why do we mirror these? Can't we just import them from where they are defined?

Reply:
They're not publicly accessible from kotlin.time.Duration. 😭

https://github.com/JetBrains/kotlin/blob/f2fa048b832148119b74e68d3cf89bf94760ad65/libraries/stdlib/src/kotlin/time/Duration.kt#L1037